src/main/java/art/heredium/service/PostService.java

```diff
@@ -186,4 +196,226 @@ private String moveEditorContent(final String editorContent, final String newPat
+    if (additionalInfo == null) return;
+
+    if (additionalInfo.getFutureExhibitionCount() != null)
+      post.setFutureExhibitionCount(additionalInfo.getFutureExhibitionCount());
+    if (additionalInfo.getOngoingExhibitionCount() != null)
+      post.setOngoingExhibitionCount(additionalInfo.getOngoingExhibitionCount());
+    if (additionalInfo.getCompletedExhibitionCount() != null)
+      post.setCompletedExhibitionCount(additionalInfo.getCompletedExhibitionCount());
+    if (additionalInfo.getFutureProgramCount() != null)
+      post.setFutureProgramCount(additionalInfo.getFutureProgramCount());
+    if (additionalInfo.getOngoingProgramCount() != null)
+      post.setOngoingProgramCount(additionalInfo.getOngoingProgramCount());
+    if (additionalInfo.getCompletedProgramCount() != null)
+      post.setCompletedProgramCount(additionalInfo.getCompletedProgramCount());
+  }
+
+  private void updateMemberships(Post post, List<PostMembershipUpdateRequest> membershipRequests) {
+    if (membershipRequests == null) return;
+
+    for (PostMembershipUpdateRequest membershipRequest : membershipRequests) {
+      if (membershipRequest.getId() != null) {
+        Membership membership =
+            post.getMemberships().stream()
+                .filter(m -> m.getId().equals(membershipRequest.getId()))
+                .findFirst()
+                .orElseThrow(() -> new ApiException(ErrorCode.MEMBERSHIP_NOT_FOUND));
+        updateMembership(membership, membershipRequest);
+      } else {
+        createNewMembership(post, membershipRequest);
+      }
+    }
+  }
+
+  private void updateMembership(Membership membership, PostMembershipUpdateRequest request) {
+    if (request.getName() != null) membership.setName(request.getName());
+    if (request.getPrice() != null) membership.setPrice(request.getPrice());
+    if (request.getImageUrl() != null) {
+      membershipService.validateImage(request.getImageUrl());
+      String newMembershipPath = FilePathType.MEMBERSHIP.getPath() + "/" + membership.getId();
+      String permanentImageUrl =
+          membershipService.moveImageToNewPlace(request.getImageUrl(), newMembershipPath);
+      membership.setImageUrl(permanentImageUrl);
+    }
+    if (request.getPeriod() != null) membership.setPeriod(request.getPeriod());
+
+    membershipRepository.save(membership);
+
+    updateCoupons(membership, request.getCoupons());
+  }
+
+  private void createNewMembership(Post post, PostMembershipUpdateRequest request) {
+    MembershipCreateRequest createRequest = new MembershipCreateRequest();
+    createRequest.setName(request.getName());
+    createRequest.setPrice(request.getPrice());
+    createRequest.setImageUrl(request.getImageUrl());
+    createRequest.setCoupons(convertToMembershipCouponCreateRequests(request.getCoupons()));
+
+    List<MembershipCreateRequest> createRequests = List.of(createRequest);
+    membershipService.createMemberships(post.getId(), createRequests);
```

Comment:
```suggestion
    membershipService.createMemberships(post.getId(), Arrays.asList(createRequest));
```
1. List.of chỉ có từ java 9 nên ko dùng dc
2. Ko cần tạo biến mới, nên lồng luôn 2 trong 1 để code đỡ bị dài.

Reply:
I updated in https://github.com/Incuvers-IT/heredium-BE/pull/68/commits/c5e4bdc0f2a7401e4ddd733dc5e5793f4c9540e0